### PR TITLE
Feature/jenkins env update

### PR DESCRIPTION
**Pull request by ameenhaq**
* Moved it folder to main directory - separating integration tests and unit tests
* Moved sbt it tests to parallel stage in analysis stage
* Created scala sub-directory for Scala it tests under it/..
* Added path to sbt home - allows use of actual 'sbt' command
* .... and other minor changes

### CODE REVIEW

**Comment by willis7** [comment]
"Updated Jenkinsfile" is not an acceptable commit message - I ended up fixing all of these myself last time. They add no value as the reader has no idead what has changes.

Wouldn't it have made sense to make [these](https://github.com/ONSdigital/sbr-control-api/pull/58) fixes at the same time?

File: Jenkinsfile [hunk 1]
```diff
@@ -176,8 +179,8 @@ pipeline {
             }
             steps {
                 sh """
-                    $SBT clean compile package
-                    $SBT clean compile assembly
+                    sbt clean compile package
+                    sbt clean compile assembly
```
Comment (willis7) [inline Jenkinsfile]: Is this duplication necessary?

File: Jenkinsfile [hunk 2]
```diff
@@ -196,20 +203,20 @@ pipeline {
             }
         }
 
-        stage('Integration Tests') {
+        stage('End to End Test') {
```
[the remark this answers is not in the document]
Comment (ameenhaq) [reply Jenkinsfile]: I've removed this stage as its not used. I've also incorporated integration stage as part of the parallel tests - in Statistical Analysis. If the name change is applicable and @nigelhp is okay with it I'm happy to rename.

File: Jenkinsfile [hunk 3]
```diff
@@ -250,3 +257,5 @@ def deploy () {
         deployToCloudFoundry("${TEAM}-${env.DEPLOY_NAME}-cf", "${CF_ORG}", "${CF_SPACE}", "${env.DEPLOY_NAME}-${MODULE_NAME}", "${env.DEPLOY_NAME}-${ORGANIZATION}-${MODULE_NAME}.zip", "conf/${env.DEPLOY_NAME}/manifest.yml")
     }
 }
+
+def isBranch(String branchName) { env.BRANCH_NAME.toString().equals(branchName) }
```
[the remark this answers is not in the document]
Comment (ameenhaq) [reply Jenkinsfile]: I see this as a temporary function until the plugin is properly fixed, I believe there is a PR being reviewed for it.

File: build.sbt [hunk 4]
```diff
@@ -48,9 +48,9 @@ lazy val Resolvers: Seq[MavenRepository] = Seq(
 )
 
 lazy val testSettings: Seq[Def.Setting[_]] = Seq(
-  sourceDirectory in ITest := baseDirectory.value / "test/it",
+  sourceDirectory in ITest := baseDirectory.value / "it",
   resourceDirectory in ITest := baseDirectory.value / "test/resources",
-  scalaSource in ITest := baseDirectory.value / "test/it",
+  scalaSource in ITest := baseDirectory.value / "it",
```
[the remark this answers is not in the document]
Comment (ameenhaq) [reply build.sbt]: Not sure about using this convention although I am aware of it. Maybe @nigelhp can offer some input in this. I have however added a sub-directory under the base directory it - 'scala' as is best practice.

File: Jenkinsfile (hunk 1 above)
Comment (ameenhaq) [reply Jenkinsfile]: Yes you're right, although neither might be needed. This depends on whether we wish to archive the .jar or packageBin zip (the latter of which is only needed for deployment/ redeployment) to Artifactory. However I believe @awharris did want the assembly jar to be archived - may need to reopen this dicussion.

File: build.sbt (hunk 4 above)
[the remark this answers is not in the document]
Comment (nigelhp) [reply build.sbt]: Unfortunately Play by default does not use either the Java or Scala standard project layout (originally inherited from Maven) e.g. Play projects do not even have a src directory.

I would not try to enforce this convention in Play projects.
Play has a /test folder by default for unit-tests.  Acceptance tests typically go in /it.

[the remark this answers is not in the document]
Comment (willis7) [reply build.sbt]: If Play has it's own convention, then it's best to follow that :+1:

File: Jenkinsfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (willis7) [reply Jenkinsfile]: Maybe a comment explaining this is a temporary  fix until the PR is merged will suffice then?

**Comment by willis7** [comment]
I just spotted we are using the squash and merge strategy, so the individual commits are less important. Approved 🆗

**Pull request closed without merging**